README.md

```diff
@@ -1,116 +1,144 @@
-  if n == 0
-    return 100
-  elsif n == -1
-    return 200
-  end
-  if n%2 == 0
-    return mystery3(n/2)
+- _Added Fun: How could we make `mystery2(-123)` work the way we might expect it to work instead of the way it does?
+  def mystery2_new(n)
+  if n.abs < 10
+    return n
   else
-    return mystery3(n-1)
+    c = (n<0 ? -1 : 1)
+    return c * (n.abs%10) + mystery2_new(c*(n.abs/10))
   end
 end
-```
+    
 
+### Trace #3
 - What is mystery3(1)?
+  Answer = 100
+
 - What is mystery3(13)?
+  Answer = 100
+
 - What is mystery3(-6)?
+  Answer = 200
+
 
-### Trace #4
-```
-def mystery4(b,e)
-  if e == 0
-    return 1
-  else
-    return b * mystery4(b,e-1)
-  end
-end
-```
 
+### Trace #4
 - What is mystery4(10,2)?
+ Answer = 1*10*10 = 100
+
 - What is mystery4(4,3)?
+  Answer = 1*4*4*4 = 64
+
 - What is mystery4(5,0)?
+  Answer = 1
 
-### Trace #5
-```
-def mystery5(s)
-  if s.length == 0
-    return ""
-  else
-    return "*" + mystery5(s[1..-1])
-  end
-end
-```
 
+### Trace #5
 - What is mystery5("hi")?
+  mystery5(“hi”) = “*” + mystery5(“i”)
+  mystery5("i”) = “*” + mystery5(“”)
+  mystery5(“") = “""
+  Answer is “**”
+
 - What is mystery5("")?
+  Answer is ""
+
 - What is mystery5("Hi, there!")?
+mystery5("Hi, there!”) = “*” + mystery5("i, there!”)
+mystery5("i, there!”) = “*” + mystery5(", there!”)
+mystery5(", there!”) = “*” + mystery5(“ there!”)
+mystery5("there!”) = “*” + mystery5("here!”)
+mystery5("here!”)  = “*” + mystery5("ere!”)
+mystery5("ere!”) = “*” + mystery5("re!”)
+mystery5("re!”) = “*” + mystery5("e!”)
+mystery5(“e!”) = “*” + mystery5("!”)
+mystery5("!”) = “*” + mystery5(“")
+mystery5(“") = "'
+ Answer is “**********"
+
 - _Added Fun: How could we make only alphabetic characters to be changed to stars?_
 
-### Trace #6
-```
-def mystery6(s)
-  if s == nil || s.length == 0
+  def mystery5_new(s)
+  if s.length == 0
     return ""
   else
-    space = 0
-    until space >= s.length || s[space] == " "
-      space += 1
-    end
-    return mystery6(s[(space+1)..-1]) + " " + s[0...space]
+    c = ("a".."z").include?(s[0].downcase) ? "*" : s[0]
+    return c + mystery5_new(s[1..-1])  
+  end  
   end
-end
-```
 
+### Trace #6
 - What is mystery6("goodnight moon")?
+  mystery6("goodnight moon")= mystery6("goodnight") + " " + "goodnight"
+  mystery6("moon") = "" + "" + "moon"
+  Answer = "moon goodnight"
+
 - What is mystery6("Ada Developers Academy")?
+ mystery6("Ada Developers Academy") = mystery6("Developers Academy") + " " + "Ada"
+ mystery6("Developers Academy") = mystery6("Academy") + " " + "Developers"
+ mystery6("Academy") = "" + "" + "Academy"
+ Answer is "Academy Developers Ada"
+
 - What is mystery6("Hi, there!")?
-- _Added Fun: How could we make the reversal happen by letter, instead of by word (i.e. Make it so that mystery6("goodnight moon") returned "noom thgindoog")?_
+  mystery6("Hi, there!") =   mystery6("there!") + " " + "Hi"
+  mystery6("there!") = "" + " " + "there!"
+  Answer = "there! Hi,"
```

Comment:
very minor thing, but all these answers are missing a leading space because of how the method is written.

e.g. mystery6("goodnight moon") should result in " moon goodnight"